version: '3.8'
services:
  backend:
    build:
      context: ./Personal_finance_tracker  
    ports:
      - "8080:8080"
    environment:
      DB_HOST: mysql-db
      DB_PORT: 3306
      DB_USERNAME: root
      DB_PASSWORD: Neshtorandom268#
    depends_on:
      mysql-db:
        condition: service_healthy
    restart: always
    networks:
      - my_network  

  mysql-db:
    image: matejgorjanov201536/personal_finance_tracker:mysql
    ports:
      - "3306:3306"
    environment:
      MYSQL_ROOT_PASSWORD: Neshtorandom268#
      MYSQL_DATABASE: financetrackerschema
      MYSQL_USER: secondaryUser          
      MYSQL_PASSWORD: Neshtorandom268#
    volumes:
      - mysql_data:/var/lib/mysql
      - C:/Users/Asus/Desktop/DevOps Project/db/init:/docker-entrypoint-initdb.d
    networks:
      - my_network
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost", "-p${MYSQL_ROOT_PASSWORD}"]
      interval: 10s
      timeout: 5s
      retries: 3

  frontend:
    build:
      context: ./Frontend
    ports:
      - "3000:3000"
    volumes:
      - ./Frontend:/app           
      - /app/node_modules         
    environment:
      NODE_ENV: development
    networks:
      - my_network


volumes:
  mysql_data:

networks:
  my_network:
    driver: bridge
